From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Thu, 19 Nov 2009 15:44:01 +0100
Subject: [PATCH] go7007: Revert reference to missing include

As part of the laudable and required clean up of the staging go7007 webcam
driver, external declarations of s2250loader_init() and
s2250loader_cleanup() got factored out to the new s2250-loader.h without
actually providing this new header and therefore breaking compilation of
VIDEO_GO7007_USB_S2250_BOARD, reported as:
- http://lkml.indiana.edu/hypermail/linux/kernel/0909.3/00062.html
- http://lkml.indiana.edu/hypermail/linux/kernel/0911.1/02303.html

This partially reverts

Commit:	fd9a40da1db372833e1af6397d2f6c94ceff3dad
	V4L/DVB (12859): go7007: semaphore -> mutex conversion

and removes obviously broken (without the accompanying s2250-loader.h)
hunks unrelated to the commit message.

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
I'm hereby CCing Linus directly, as it is late in the -rc cycle and fails
to compile since september 21st and is reported since the 24th.
Compile-tested only, as I don't own the corresponding hardware, but it is
a strict revert to the situation in 2.6.31. The alternatives would be to
revert fd9a40da1db372833e1af6397d2f6c94ceff3dad as a whole or to mark
VIDEO_GO7007_USB_S2250_BOARD as BROKEN.

 drivers/staging/go7007/s2250-board.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/drivers/staging/go7007/s2250-board.c b/drivers/staging/go7007/s2250-board.c
index 8c85a9c..0de09a2 100644
--- a/drivers/staging/go7007/s2250-board.c
+++ b/drivers/staging/go7007/s2250-board.c
@@ -21,10 +21,12 @@
 #include <linux/i2c.h>
 #include <linux/videodev2.h>
 #include <media/v4l2-common.h>
-#include "s2250-loader.h"
 #include "go7007-priv.h"
 #include "wis-i2c.h"
 
+extern int s2250loader_init(void);
+extern void s2250loader_cleanup(void);
+
 #define TLV320_ADDRESS      0x34
 #define VPX322_ADDR_ANALOGCONTROL1	0x02
 #define VPX322_ADDR_BRIGHTNESS0		0x0127
-- 
1.6.5.3

